src/main/java/com/amazon/opendistro/opensearch/performanceanalyzer/metricsdb/MetricsDB.java

```diff
@@ -0,0 +1,421 @@
+        create.insertInto(DSL.table(metric.getName()))
+                .set(DSL.field(SUM, Double.class), metric.getSum())
+                .set(DSL.field(AVG, Double.class), metric.getAvg())
+                .set(DSL.field(MIN, Double.class), metric.getMin())
+                .set(DSL.field(MAX, Double.class), metric.getMax())
+                .set(dimensions.getFieldMap())
+                .execute();
+    }
+
+    /**
+     * Drop a metric table. This is for IT framework to use only
+     *
+     * @param metricName metric table to be deleted
+     */
+    @VisibleForTesting
+    public void deleteMetric(String metricName) {
+        if (DBUtils.checkIfTableExists(create, metricName)) {
+            create.dropTable(metricName).execute();
+        }
+    }
+
+    // We have a table per metric. We do a group by/aggregate on
+    // every dimension and return all the metric tables.
+    public List<TableLike<Record>> getAggregatedMetricTables(
+            List<String> metrics, List<String> aggregations, List<String> dimensions)
+            throws Exception {
+        List<TableLike<Record>> tList = new ArrayList<>();
+        List<Field<?>> groupByFields = DBUtils.getFieldsFromList(dimensions);
+
+        for (int i = 0; i < metrics.size(); i++) {
+            String metric = metrics.get(i);
+            List<Field<?>> selectFields = DBUtils.getFieldsFromList(dimensions);
+            String aggType = aggregations.get(i);
+            if (aggType.equals(SUM)) {
+                Field<Double> field = DSL.field(SUM, Double.class);
+                selectFields.add(DSL.sum(field).as(metric));
+            } else if (aggType.equals(AVG)) {
+                Field<Double> field = DSL.field(AVG, Double.class);
+                selectFields.add(DSL.avg(field).as(metric));
+            } else if (aggType.equals(MIN)) {
+                Field<Double> field = DSL.field(MIN, Double.class);
+                selectFields.add(DSL.min(field).as(metric));
+            } else if (aggType.equals(MAX)) {
+                Field<Double> field = DSL.field(MAX, Double.class);
+                selectFields.add(DSL.max(field).as(metric));
+            } else {
+                throw new Exception("Unknown agg type");
+            }
+            if (!DBUtils.checkIfTableExists(create, metrics.get(i))) {
+                tList.add(null);
+            } else {
+                tList.add(
+                        create.select(selectFields)
+                                .from(DSL.table(metric))
+                                .groupBy(groupByFields)
+                                .asTable());
+            }
+        }
+        return tList;
+    }
+
+    /**
+     * query metrics from different tables and merge to one table.
+     *
+     * <p>getAggregatedMetricTables returns tables like: +-----+---------+-----+ |shard|indexName|
+     * cpu| +-----+---------+-----+ |0 |sonested | 10| |1 |sonested | 20|
+     *
```

Comment:
Spotless may have messed with your manual formatting. You can use `spotless:off` and `spotless:on` annotations to make spotless ignore sections:

```
/**
 * This is a table
 *
 * spotless:off
 * |column1|column2|
 * |    100|    200|
 * spotless:on
 */